crates/biome_js_formatter/src/js/expressions/call_arguments.rs

```diff
@@ -667,12 +668,43 @@ fn with_token_tracking_disabled<F: FnOnce(&mut JsFormatter) -> R, R>(
     result
 }
 
-/// Tests if `expression` has an empty parameters list.
-fn has_no_parameters(expression: &JsFunctionExpression) -> bool {
-    match expression.parameters() {
-        // Use default formatting for expressions without parameters, will return `Err` anyway
-        Err(_) => true,
-        Ok(parameters) => parameters.items().is_empty(),
+/// Tests if all of the parameters of `expression` are simple enough to allow
+/// a function to group.
+fn has_only_simple_parameters(expression: &JsFunctionExpression) -> bool {
+    expression.parameters().map_or(true, |parameters| {
+        for parameter in parameters.items().into_iter().flatten() {
+            if !is_simple_parameter(parameter) {
+                return false;
+            }
+        }
+
+        true
+    })
+}
+
+/// Tests if the single parameter is "simple", as in a plain identifier with no
+/// explicit type annotation and no initializer:
+///
+/// Examples:
+/// foo             => true
+/// foo?            => true
+/// foo = 'bar'     => false
+/// foo: string     => false
+/// {a, b}          => false
+/// {a, b} = {}     => false
+/// [a, b]          => false
+///
+fn is_simple_parameter(parameter: AnyJsParameter) -> bool {
```

Comment:
I don't think we need to take ownership here.
```suggestion
fn is_simple_parameter(parameter: &AnyJsParameter) -> bool {
```